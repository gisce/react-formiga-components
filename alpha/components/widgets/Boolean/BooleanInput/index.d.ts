export * from './BooleanInput';
//# sourceMappingURL=index.d.ts.map