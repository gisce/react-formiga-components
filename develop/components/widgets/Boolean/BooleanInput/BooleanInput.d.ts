import { BaseFieldProps } from '../../../../components/form';
export declare const BooleanInput: (props: BaseFieldProps<boolean>) => import("react/jsx-runtime").JSX.Element;
//# sourceMappingURL=BooleanInput.d.ts.map