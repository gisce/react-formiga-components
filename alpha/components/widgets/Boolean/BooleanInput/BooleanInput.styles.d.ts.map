{"version":3,"file":"BooleanInput.styles.d.ts","sourceRoot":"","sources":["../../../../../src/components/widgets/Boolean/BooleanInput/BooleanInput.styles.tsx"],"names":[],"mappings":"AAGA,eAAO,MAAM,gBAAgB,EAAE,GAQ9B,CAAC;AAEF,eAAO,MAAM,iBAAiB,oEAG7B,CAAC"}