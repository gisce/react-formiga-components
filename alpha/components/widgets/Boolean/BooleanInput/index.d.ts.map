{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../../../src/components/widgets/Boolean/BooleanInput/index.ts"],"names":[],"mappings":"AAAA,cAAc,gBAAgB,CAAC"}