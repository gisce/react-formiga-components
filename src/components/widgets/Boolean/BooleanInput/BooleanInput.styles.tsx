import { Checkbox as AntCheckbox } from "antd";
import styled from "styled-components";

export const RequiredCheckbox: any = styled(AntCheckbox)`
  .ant-checkbox-inner {
    background-color: #dbeafe;
  }

  .ant-checkbox-checked .ant-checkbox-inner {
    background-color: #089cff;
  }
`;

export const CheckboxContainer = styled.div`
  display: flex;
  flex-direction: row;
`;