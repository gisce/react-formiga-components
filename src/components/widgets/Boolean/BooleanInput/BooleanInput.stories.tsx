import React, { useEffect, useState } from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { BooleanInput } from "./BooleanInput";
import { BaseFieldArgTypes } from "components/form/Field/BaseField.argTypes";

export default {
  title: "Components/Widgets/Boolean/BooleanInput",
  component: BooleanInput,
  argTypes: {
    ...BaseFieldArgTypes,
    locale: { table: { disable: true } },
  },
} as ComponentMeta<typeof BooleanInput>;

const Template: ComponentStory<typeof BooleanInput> = (args) => {
  const [value, setValue] = useState(args.value);

  useEffect(() => {
    setValue(args.value);
  }, [args.value]);

  return (
    <BooleanInput
      {...args}
      value={value}
      onChange={(value: boolean) => {
        setValue(value);
        args.onChange?.(value);
      }}
    />
  );
};

export const Basic = Template.bind({});
Basic.args = {
  required: false,
  readOnly: false,
  value: false,
};
