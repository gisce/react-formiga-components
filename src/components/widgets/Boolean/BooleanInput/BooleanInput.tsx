import React from "react";
import { Checkbox as AntCheckbox } from "antd";
import { CheckboxContainer, RequiredCheckbox } from "./BooleanInput.styles";
import { BaseFieldProps } from "@/components/form";

export const BooleanInput = (props: BaseFieldProps<boolean>) => {
  const { required, readOnly, value, onChange, ...restProps } = props;

  const CustomCheckbox: any =
    required && !readOnly ? RequiredCheckbox : AntCheckbox;

  return (
    <CheckboxContainer data-testid="checkbox_container">
      <CustomCheckbox
        {...restProps}
        data-testid="checkbox_input"
        disabled={readOnly}
        checked={value}
        onChange={(event: { target: { checked: boolean } }) => {
          onChange(event.target.checked);
        }}
      />
    </CheckboxContainer>
  );
};
