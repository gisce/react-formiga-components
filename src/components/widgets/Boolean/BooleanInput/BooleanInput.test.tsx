import React from "react";
import { render } from "@testing-library/react";

import { BooleanInput } from "./BooleanInput";
import { BaseFieldProps } from "index";
import { composeStories } from "@storybook/testing-react";
import * as stories from "./BooleanInput.stories";

describe("BooleanInput", () => {
  const testCases = Object.values(composeStories(stories)).map((Story) => [
    Story.storyName!,
    Story,
  ]);

  test.each(testCases)("Renders %s story", async (_storyName, Story) => {
    const tree = render(<Story />);
    expect(tree.baseElement).toMatchSnapshot();
  });

  const renderComponent = ({
    readOnly,
    required,
    value,
    onChange,
  }: Partial<BaseFieldProps<boolean>>) =>
    render(
      <BooleanInput
        required={required}
        readOnly={readOnly}
        value={value}
        onChange={onChange}
      />
    );

  it("should render a checkbox correctly", () => {
    const { getByTestId } = renderComponent({});
    const testComponent = getByTestId("checkbox_input");
    expect(testComponent).not.toBeDisabled();
    expect(testComponent).not.toBeChecked();
    expect(testComponent).toHaveAttribute("type", "checkbox");

    const checkbox_container = getByTestId("checkbox_container");
    const innerSpan = checkbox_container.querySelector(".ant-checkbox-inner");
    expect(innerSpan).not.toHaveStyle("background-color: #dbeafe;");
  });

  it("should render a required checkbox correctly", () => {
    const { getByTestId } = renderComponent({
      required: true,
    });

    const testComponent = getByTestId("checkbox_input");
    expect(testComponent).not.toBeDisabled();
    expect(testComponent).not.toBeChecked();
    expect(testComponent).toHaveAttribute("type", "checkbox");

    const checkbox_container = getByTestId("checkbox_container");
    const innerSpan = checkbox_container.querySelector(".ant-checkbox-inner");
    expect(innerSpan).toHaveStyle("background-color: #dbeafe;");
  });

  it("should render a read only checkbox correctly", () => {
    const { getByTestId } = renderComponent({
      readOnly: true,
    });

    const testComponent = getByTestId("checkbox_input");
    expect(testComponent).toBeDisabled();
    expect(testComponent).not.toBeChecked();
    expect(testComponent).toHaveAttribute("type", "checkbox");
  });

  it("should render a checkbox checked correctly", () => {
    const { getByTestId } = renderComponent({ value: true });
    const testComponent = getByTestId("checkbox_input");
    expect(testComponent).not.toBeDisabled();
    expect(testComponent).toBeChecked();
    expect(testComponent).toHaveAttribute("type", "checkbox");
  });

  it("should render a checkbox unchecked correctly", () => {
    const { getByTestId } = renderComponent({ value: false });
    const testComponent = getByTestId("checkbox_input");
    expect(testComponent).not.toBeDisabled();
    expect(testComponent).not.toBeChecked();
    expect(testComponent).toHaveAttribute("type", "checkbox");
  });

  it("should render a checkbox unchecked when clicked", () => {
    const onChangeFn = jest.fn();
    const { getByTestId } = renderComponent({ onChange: onChangeFn });
    const testComponent = getByTestId("checkbox_input");
    expect(testComponent).not.toBeDisabled();
    expect(testComponent).not.toBeChecked();
    expect(testComponent).toHaveAttribute("type", "checkbox");

    testComponent.click();
    expect(onChangeFn).toHaveBeenCalledTimes(1);
    expect(onChangeFn).toHaveBeenCalledWith(true);
  });

  it("should render a checkbox checked when clicked", () => {
    const onChangeFn = jest.fn();
    const { getByTestId } = renderComponent({
      value: true,
      onChange: onChangeFn,
    });
    const testComponent = getByTestId("checkbox_input");
    expect(testComponent).not.toBeDisabled();
    expect(testComponent).toBeChecked();
    expect(testComponent).toHaveAttribute("type", "checkbox");

    testComponent.click();
    expect(onChangeFn).toHaveBeenCalledTimes(1);
    expect(onChangeFn).toHaveBeenCalledWith(false);
  });
});
