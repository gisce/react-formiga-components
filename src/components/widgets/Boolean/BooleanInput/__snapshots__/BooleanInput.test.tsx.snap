// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooleanInput Renders Basic story 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<body>
  <div>
    <div
      class="c0"
      data-testid="checkbox_container"
    >
      <label
        class="ant-checkbox-wrapper"
      >
        <span
          class="ant-checkbox"
        >
          <input
            class="ant-checkbox-input"
            data-testid="checkbox_input"
            type="checkbox"
            value=""
          />
          <span
            class="ant-checkbox-inner"
          />
        </span>
      </label>
    </div>
  </div>
</body>
`;
